src/main/java/com/lambdaworks/redis/resource/Delay.java

```diff
@@ -51,46 +58,159 @@ public TimeUnit getTimeUnit() {
     public abstract long createDelay(long attempt);
 
     /**
-     * Creates a new {@link ConstantDelay}.
+     * Reset this delay.
+     *
+     * This method is to be implemented by the implementations.
+     */
+    public void reset() {
```

Comment:
Let's move that into an own interface. By that way we can discover on the `ClientResources` side whether a `Delay` is stateful and reject `reconnectDelay(Delay)` calls to enforce a `reconnectDelay(Supplier<Delay>)`.